### PR TITLE
Add an option to change the default shell.

### PR DESCRIPTION
**Pull request by meator**
This adds an option to set the default shell of the `root` user and documents the `live.shell` kernel option.

I know that a similar change has already been closed (#99). My change makes this more configurable and optional. And it documents the `live.shell` option that isn't documented anywhere else and is pretty useful.

I'll understand if you close this, but I consider this helpful for people who are customizing their iso.

### CODE REVIEW

**Comment by Vaelatern** [comment]
I do not object to this.

Configuration in this particular area might be useful. The previous PR posited "The default shell should be bash" which was wrong. This one posits "The default shell should be configurable on image creation" which seems reasonable.

I'm pretty sure the commit message is incorrect.

**Comment by meator** [comment]
I've made the commit message clearer.
I've also added a check that increases `STEP_COUNT` by one when the `-S` option is used.

File: mklive.sh.in [hunk 1]
```diff
@@ -75,6 +75,8 @@ directory if unset).
  -l <locale>        Default locale to use (en_US.UTF-8 if unset).
  -i <lz4|gzip|bzip2|xz> Compression type for the initramfs image (xz if unset).
  -s <gzip|lzo|xz>     Compression type for the squashfs image (xz if unset)
+ -S <shell>         Default shell of the root user (must be absolute path). 
+                    Set the live.shell kernel parameter to change the default shell of anon.
```
Comment (0x5c) [inline mklive.sh.in:79]: Since this PR was made, https://github.com/void-linux/void-mklive/pull/267 was merged and added a -S argument.

**Review by 0x5c — COMMENTED** [review]
This would be a neat feature to have

The option doesn't currently set the shell for the `anon` user, should this also be implemented? I think it might be doable by modifying `dracut/vmklive/adduser.sh` when creating the image.

File: mklive.sh.in (hunk 1 above)
Comment (meator) [reply mklive.sh.in]: I changed the flag to `-e`. The flag letters are running out.

**Review by classabbyamp — APPROVED** [review]

**Pull request merged**